tests/Integration/Behaviour/Features/Context/Domain/Product/Combination/UpdateCombinationSuppliersFeatureContext.php

```diff
@@ -0,0 +1,163 @@
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain\Product\Combination;
+
+use Behat\Gherkin\Node\TableNode;
+use Currency;
+use PHPUnit\Framework\Assert;
+use PrestaShop\Decimal\DecimalNumber;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Command\RemoveAllAssociatedCombinationSuppliersCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Command\SetCombinationSuppliersCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Query\GetCombinationSuppliers;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Supplier\QueryResult\ProductSupplierInfo;
+use PrestaShop\PrestaShop\Core\Domain\Product\Supplier\ValueObject\ProductSupplierId;
+
+class UpdateCombinationSuppliersFeatureContext extends AbstractCombinationFeatureContext
+{
+    /**
+     * @When I set following suppliers for combination ":combinationReference":
+     *
+     * @param string $combinationReference
+     * @param TableNode $table
+     */
+    public function setCombinationSuppliers(string $combinationReference, TableNode $table): void
+    {
+        $references = [];
+        $productSuppliers = [];
+        foreach ($table->getColumnsHash() as $row) {
+            $productSupplierId = null;
+            $references[] = $row['reference'];
+            if ($this->getSharedStorage()->exists($row['reference'])) {
+                $productSupplierId = $this->getSharedStorage()->get($row['reference']);
+            }
+
+            $productSuppliers[] = [
+                'supplier_id' => $this->getSharedStorage()->get($row['supplier reference']),
+                'currency_id' => (int) Currency::getIdByIsoCode($row['currency'], 0, true),
+                'reference' => $row['combination supplier reference'],
+                'price_tax_excluded' => $row['price tax excluded'],
+                'combination_id' => $this->getSharedStorage()->get($combinationReference),
+                'product_supplier_id' => $productSupplierId,
+            ];
+        }
+
+        $command = new SetCombinationSuppliersCommand(
+            $this->getSharedStorage()->get($combinationReference),
+            $productSuppliers
+        );
+
+        $productSupplierIds = $this->getCommandBus()->handle($command);
+
+        Assert::assertSameSize(
+            $references,
+            $productSupplierIds,
+            'Cannot set references in shared storage. References and actual combination suppliers doesn\'t match.'
+        );
+
+        /** @var ProductSupplierId $productSupplierId */
+        foreach ($productSupplierIds as $key => $productSupplierId) {
+            $this->getSharedStorage()->set($references[$key], $productSupplierId->getValue());
+        }
+    }
+
+    /**
+     * @When I remove all associated combination ":combinationReference" suppliers
```

Comment:
```suggestion
     * @When I mark combination ":combinationReference" is not provided by suppliers anymore
```
Following previous idea?